### PR TITLE
Still support the latest older versions of doctrines orm 2, dbal 3 and lexer 3

### PR DESCRIPTION
**Pull request by alexander-schranz**
After lexer 3 support is merged. I rebased this branch to keep support for older doctrine versions.

# TODO

- [x] Run tests?

### CODE REVIEW

**Comment by x86demon** [comment]
@alexander-schranz to run tests follow steps defined in the `.travis.yml`

https://github.com/oroinc/doctrine-extensions/blob/master/.travis.yml#L27-L42

**Comment by alexander-schranz** [comment]
I did not yet have any time to test if all test still runs. Please feel free to take this over and check all the test matrixes defined in the .travis.yml. For @x86demon I would recommend to switch to Github Actions as travis CI is not longer run here for the pull requests as I think travis did cancel open source support in [August 2022](https://travis-ci.community/t/question-open-source-travis-ci-for-open-source-projects-as-of-august-2022/13233)

**Comment by x86demon** [comment]
@alexander-schranz thank you for recommendation of Github Actions, I'll take a look.

**Comment by monpetitplacement** [comment]
I made this request in the PR https://github.com/oroinc/doctrine-extensions/pull/99

**Comment by Gwemox** [comment]
I made this request in the PR https://github.com/oroinc/doctrine-extensions/pull/99

**Comment by alexander-schranz** [comment]
We are currently stuck on `orm`: `2` would still be great to have a version of oro extensions which supports the lexer 3 also with orm 2. I think there is atleast one `2.x` doctrine/orm version which is fully compatibility with `doctrine/orm` `3.x` as they have a smooth upgrade and deprecation path inside doctrine.

**Comment by Gwemox** [comment]
Your code works with Lexer 2.0 and Lexer 3.0, however the tests fail because you need to change the annotation with `doctrine/annotations: 2.0` :

**tests/Oro/Tests/Connection/TestUtil.php:**
```php
$config = Setup::createAnnotationMetadataConfiguration([$entitiesPath], true);
```
should be changed to
```php
$config = Setup::createAnnotationMetadataConfiguration([$entitiesPath], true, null, null, false);
```

**tests/Oro/Entities/Foo.php:**
Change the annotations to use the prefix `@ORM\` and include:
```php
use Doctrine\ORM\Mapping as ORM;
```

After these changes, I think we can merge into version 2.x.

**Comment by Gwemox** [comment]
> @alexander-schranz thank you for recommendation of Github Actions, I'll take a look. Please make sure your changes are compatible with both versions of the lexer. Now it looks like a change in favor of v3 only.

New CI here : https://github.com/oroinc/doctrine-extensions/pull/100

**Comment by alexander-schranz** [comment]
I rebased and adopted the Test Matrix to show that even on the new prefer-lowest the tests seems to run as expected.

**Comment by Gwemox** [comment]
The tests do not explicitly verify proper functioning with Doctrine version 3 or 2. In case they are blocked by another dependency. Wouldn't it be better to use the --with parameter of Composer?

**Comment by alexander-schranz** [comment]
@Gwemox sure we could make a more complex CI but still `--prefer-lowest` should always part of a CI to make sure that version constraints are correctly defined :). What we could is also add a test to check if `--prefer-lowest` runs on expected versions instead of add more and more CI builts.

I did have a quicklook at doctrine/dbal 2 support. But think that is something which we should skip. That are too many BC layer changes that it is worth.

File: tests/lowest-dependencies.php
Comment (alexander-schranz) [inline tests/lowest-dependencies.php:1]: This make sure that our CI not accidently not longer tests dbal 3, orm 2 or lexer 2.

**Comment by alexander-schranz** [comment]
2.0.5 add support for lexer 3

**Comment by alexander-schranz** [comment]
@x86demon did let the CI rerun. Looks all fine. Can you recheck this Pull Request?

**Comment by x86demon** [comment]
@alexander-schranz PR looks good, thank you for contribution, Is it ready for merge?

**Comment by alexander-schranz** [comment]
@x86demon Thx. Is something else we need to tackle for the 3.0 release?

**Pull request merged**